server/utils/user-stats.js

```diff
@@ -1,46 +1,55 @@
+import _ from 'lodash';
 import moment from 'moment-timezone';
 import { dayCount } from '../utils/date-utils';
 
 const daysBetween = 1.5;
 
-export function calcCurrentStreak(cals, timezone = 'UTC') {
-  const revCals = cals.slice().reverse();
+export function prepUniqueDays(cals, tz = 'UTC') {
 
-  if (dayCount([moment().tz(timezone), revCals[0]], timezone) > daysBetween) {
+  return _(cals)
+    .map(ts => moment(ts).tz(tz).startOf('day').valueOf())
+    .uniq()
+    .sort()
+    .value();
+}
+
+export function calcCurrentStreak(cals, tz = 'UTC') {
+
+  let prev = _.last(cals);
+  if (moment().tz(tz).startOf('day').diff(prev, 'days') > daysBetween) {
     return 0;
   }
+  let currentStreak = 0;
+  let streakContinues = true;
+  _(cals)
+    .forEachRight(cur => {
+    if (moment(prev).diff(cur, 'days') < daysBetween) {
+      prev = cur;
+      currentStreak++;
+    } else {
+      // current streak found
+      streakContinues = false;
+    }
+    return streakContinues;
+  });
 
```

Comment:
Line 26-34 should be indented to indicate they are part of the `.forEachRight` function